src/main/java/appeng/util/item/AEItemStack.java

```diff
@@ -156,41 +111,17 @@ public static IAEItemStack loadItemStackFromPacket( final ByteBuf data ) throws
 		final byte stackType = (byte) ( ( mask & 0x0C ) >> 2 );
 		final byte countReqType = (byte) ( ( mask & 0x30 ) >> 4 );
 		final boolean isCraftable = ( mask & 0x40 ) > 0;
-		final boolean hasTagCompound = ( mask & 0x80 ) > 0;
-
-		// don't send this...
-		final NBTTagCompound d = new NBTTagCompound();
-
-		// For some insane reason, Vanilla can only parse numeric item ids if they are strings
-		short itemNumericId = data.readShort();
-		d.setString( "id", String.valueOf( itemNumericId ) );
-		d.setShort( "Damage", data.readShort() );
-		d.setByte( "Count", (byte) 1 ); // 1 Because vanilla'll freak out otherwise
-
-		if( hasTagCompound )
-		{
-			final int len = data.readInt();
-
-			final byte[] bd = new byte[len];
-			data.readBytes( bd );
 
-			final ByteArrayInputStream di = new ByteArrayInputStream( bd );
-			d.setTag( "tag", CompressedStreamTools.read( new DataInputStream( di ) ) );
-		}
-
-		// long priority = getPacketValue( PriorityType, data );
+		final ItemStack itemstack = ByteBufUtils.readItemStack( data );
 		final long stackSize = getPacketValue( stackType, data );
 		final long countRequestable = getPacketValue( countReqType, data );
 
-		final ItemStack itemstack = new ItemStack( d );
-
 		if( itemstack.isEmpty() )
 		{
 			return null;
 		}
 
 		final AEItemStack item = AEItemStack.create( itemstack );
-		// item.priority = (int) priority;
 		item.setStackSize( stackSize );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "item" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2259)